lib/src/view/study/study_list_screen.dart

```diff
@@ -0,0 +1,472 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lichess_mobile/src/model/auth/auth_session.dart';
+import 'package:lichess_mobile/src/model/study/study.dart';
+import 'package:lichess_mobile/src/model/study/study_filter.dart';
+import 'package:lichess_mobile/src/model/study/study_list_paginator.dart';
+import 'package:lichess_mobile/src/styles/lichess_icons.dart';
+import 'package:lichess_mobile/src/styles/styles.dart';
+import 'package:lichess_mobile/src/utils/l10n_context.dart';
+import 'package:lichess_mobile/src/utils/lichess_assets.dart';
+import 'package:lichess_mobile/src/utils/navigation.dart';
+import 'package:lichess_mobile/src/view/study/study_screen.dart';
+import 'package:lichess_mobile/src/widgets/adaptive_bottom_sheet.dart';
+import 'package:lichess_mobile/src/widgets/buttons.dart';
+import 'package:lichess_mobile/src/widgets/filter.dart';
+import 'package:lichess_mobile/src/widgets/list.dart';
+import 'package:lichess_mobile/src/widgets/platform_scaffold.dart';
+import 'package:lichess_mobile/src/widgets/user_full_name.dart';
+import 'package:logging/logging.dart';
+import 'package:timeago/timeago.dart' as timeago;
+
+final _logger = Logger('StudyListScreen');
+
+// TODO l10n
+String studyCategoryL10n(StudyCategory category, BuildContext context) =>
+    switch (category) {
+      StudyCategory.all => 'All',
+      StudyCategory.mine => 'Mine',
+      StudyCategory.member => 'Member',
+      StudyCategory.public => 'Public',
+      StudyCategory.private => 'Private',
+      StudyCategory.likes => 'Liked',
+    };
+
+// TODO l10n
+String studyListOrderL10n(StudyListOrder order, BuildContext context) =>
+    switch (order) {
+      StudyListOrder.hot => 'Hot',
+      StudyListOrder.newest => 'Newest',
+      StudyListOrder.oldest => 'Oldest',
+      StudyListOrder.updated => 'Updated',
+      StudyListOrder.popular => 'Popular',
+    };
+
+/// A screen that displays a paginated list of studies
+class StudyListScreen extends ConsumerWidget {
+  const StudyListScreen({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isLoggedIn = ref.read(authSessionProvider)?.user.id != null;
```

Comment:
huh, it's been some time since I wrote this, but I probably assumed that the user can never change while we show this screen..? Not a good enough reason though, I changed it to `watch` now